tianshou/data/batch.py

```diff
@@ -496,6 +494,8 @@ def stack_(
         self, batches: Sequence[Union[dict, "Batch"]], axis: int = 0
     ) -> None:
         """Stack a list of Batch object into current batch."""
+        batches = [x for x in batches if isinstance(
```

Comment:
I think at this point and noting the other quite complexe comprehension loop below, maybe a good old plain for-loop would be clearer and more efficient.

Reply:
Agree. But we tend to optimize batch in the future, this is just a workaround.
Also if I don't add this line of code, what's the following result:
```python
Batch([[Batch(a=1)], [Batch(a=1)]])
```
I think it should be an illegal input, because it is not a `Sequence[Union[dict, Batch]]`.

Reply:
Hum... I don't really know, it is hard to tell for sure. It may be useful to support it, but I can't come up with an example for which it would be actually useful. So I guess it is better to forbid it if there is no relevant usecase of it.

Reply:
Yep, so that it returns an empty Batch as the temporary solution currently.

Reply:
Could you document this point please ? It would be nice to keep track of it do not forget taking this aspect into consideration for future rework.

Reply:
Emptiness sounds quite misleading, don't you think ? throwing an exception seems more appropriate no ?